sdv/multi_table/utils.py

```diff
@@ -0,0 +1,400 @@
+        relationships (list[dict]):
+            List of relationships between the tables.
+        parent_table (str):
+            Name of the parent table.
+        order (int):
+            Order of the descendants.
+    """
+    descendants = {}
+    order_1_descendants = _get_relationships_for_parent(relationships, parent_table)
+    descendants['order_1'] = [rel['child_table_name'] for rel in order_1_descendants]
+    for i in range(2, order+1):
+        descendants[f'order_{i}'] = []
+        prov_descendants = []
+        for child_table in descendants[f'order_{i-1}']:
+            order_i_descendants = _get_relationships_for_parent(relationships, child_table)
+            prov_descendants.extend([rel['child_table_name'] for rel in order_i_descendants])
+
+        descendants[f'order_{i}'] = prov_descendants
+
+    return descendants
+
+
+def _get_all_descendant_per_root_at_order_n(relationships, order):
+    """Get all the descendants of the root tables at a given order.
+
+    Args:
+        relationships (list[dict]):
+            List of relationships between the tables.
+        order (int):
+            Order of the descendants.
+    """
+    root_tables = _get_root_tables(relationships)
+    all_descendants = {}
+    for root in root_tables:
+        all_descendants[root] = {}
+        order_descendants = _get_n_order_descendants(relationships, root, order)
+        all_descendant_root = set()
+        for order_key in order_descendants:
+            all_descendant_root.update(order_descendants[order_key])
+
+        all_descendants[root] = order_descendants
+        all_descendants[root]['num_descendants'] = len(all_descendant_root)
+
+    return all_descendants
+
+
+def _simplify_relationships_and_tables(metadata, tables_to_drop):
+    """Simplify the relationships and tables of the metadata.
+
+    Removes the relationships that are not direct child or grandchild of the root table.
+    Removes the tables that are not direct child or grandchild of the root table.
+
+    Args:
+        metadata (MultiTableMetadata):
+            Metadata of the datasets.
+        tables_to_drop (set):
+            Set of the tables that relationships will be removed.
+
+    Returns:
+        MultiTableMetadata:
+            Simplified metadata.
+    """
+    relationships = deepcopy(metadata.relationships)
+    for relationship in relationships:
+        parent_table = relationship['parent_table_name']
+        child_table = relationship['child_table_name']
+        is_parent_to_drop = parent_table in tables_to_drop
+        is_child_to_drop = child_table in tables_to_drop
+        if is_parent_to_drop or is_child_to_drop:
+            metadata.relationships.remove(relationship)
+            if is_parent_to_drop and parent_table in metadata.tables:
+                del metadata.tables[parent_table]
+            elif is_child_to_drop and child_table in metadata.tables:
+                del metadata.tables[child_table]
+
+    return metadata
```

Comment:
Since this function will actually modify the metadata object we pass in, we should remove the return. The convention is usually to have functions that modify arguments in-place to not return anything.